File: additional_02_tcp_socket/TcpSocketClient.java
// Additional Experiment 2: Basic TCP Socket Client

import java.io.*;
import java.net.*;
import java.util.Scanner;

public class TcpSocketClient {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter server IP (example: 127.0.0.1): ");
        String host = scanner.nextLine().trim();

        final int PORT = 6000;

        try (
            Socket socket = new Socket(host, PORT);        // Connect to server.
            BufferedReader input = new BufferedReader(
                new InputStreamReader(socket.getInputStream())
            );
            PrintWriter output = new PrintWriter(
                socket.getOutputStream(), true
            )
        ) {
            System.out.print("Enter message: ");
            String message = scanner.nextLine();           // Read message.

            output.println(message);                        // Send message.
            String reply = input.readLine();               // Receive response.

            System.out.println("Server: " + reply);
        } catch (IOException error) {
            System.out.println("Client error: " + error.getMessage());
        }

        scanner.close();
    }
}

File: additional_02_tcp_socket/tcp_socket_client.py
# Additional Experiment 2: Basic TCP Socket Client

import socket

HOST = input("Enter server IP (example: 127.0.0.1): ").strip()
PORT = 6000

client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # Create TCP socket.
client.connect((HOST, PORT))                               # Connect to server.

message = input("Enter message: ")                         # Read data to send.
client.sendall(message.encode())                           # Send data using TCP.

reply = client.recv(1024).decode()                         # Receive server response.
print("Server:", reply)

client.close()                                             # Close TCP connection.
